galaxy_ng/app/management/commands/analytics/galaxy_collector.py

```diff
@@ -0,0 +1,220 @@
+import os
+import platform
+import distro
+from django.conf import settings
+
+from insights_analytics_collector import CsvFileSplitter, register
+
+from galaxy_ng.app.management.commands.analytics.collector import Collector
+
+
+@register("config", "1.0", description="General platform configuration.", config=True)
+def config(since, **kwargs):
+    # TODO: license_info = get_license()
+    license_info = {}
+
+    return {
+        "authentication_backends": [],
+        "platform": {
+            "system": platform.system(),
+            "dist": distro.linux_distribution(),
+            "release": platform.release(),
+            "type": "traditional",
+        },
+        "external_logger_enabled": "todo",
+        "external_logger_type": "todo",
+        "install_uuid": "todo",
+        "instance_uuid": "todo",
+        "tower_url_base": "todo",
+        "tower_version": "todo",
+        "license_expiry": "todo",
+        "logging_aggregators": ["todo"],
+        "pendo_tracking": "todo",
+        "hub_url_base": "todo",
+        "hub_version": "todo",
+        "license_type": license_info.get("license_type", "UNLICENSED"),
+        "free_instances": license_info.get("free_instances", 0),
+        "total_licensed_instances": license_info.get("instance_count", 0),
+        "license_expiry": license_info.get("time_remaining", 0),
+        "authentication_backends": settings.AUTHENTICATION_BACKENDS,
+    }
+
+
+@register("instance_info", "1.0", description="Node information", config=True)
+def config(since, **kwargs):
+    # TODO:
+
+    return {
+        "versions": {"system": "todo"},
+        "online_workers": "todo",
+        "online_content_apps": "todo",
+        "database_connection": "todo",
+        "redis_connection": "todo",
+        "storage": "todo",
+    }
+
+
+@register("ansible_collection_table", "1.0", format="csv", description="Data on ansible_collection")
+def ansible_collection_table(since, full_path, until, **kwargs):
+    source_query = """
+        COPY (
+            SELECT "ansible_collection"."pulp_id",
```

Comment:
@dmzoneill  talking with @matburt and @awcrosby , can we try to combine the models with sql here?

Something like "COPY ({model_query.sql}) TO STDOUT WITH CSV HEADER". Not sure if django models support copy natively.

The queries here are fairly simple, so using models makes sense.

Reply:
@Ladas it sounds like the ORM queries are not going to work well. And upon further thought, they might not give us many benefits in the event of a schema change. It does seem like most of these queries hit mainly one table with few joins, so maybe schema changes will not often impact these queries. So IMO the manual queries should be sufficient.

Reply:
@awcrosby yeah, in Controller we have raw queries and we had to do maybe 1 or 2 changes last couple of years, due to schema drift.

So yeah it doesn't give us many benefits. Especially if we are not able to do all queries and we'd have a mix of ORM/SQL here then. It would be good to use ORM, if we'd be exporting data that have a lot of business logic in models, but that doesn't seem to be the case. (and the business logic in models would slow this down a lot, so we'd rather replicate in DB I guess...)